### PR TITLE
[GDB-10116]-Added-Private-Link-Support

### PR DESCRIPTION
**Pull request by simeonzhekofff**
## Description

- Added support for the AWS PrivateLink service.
- Added tags for the VPC
- Added conditions for enable/disable AWS PrivateLink service.

## Related Issues

[GDB-10116]

## Changes

- Support for the AWS PrivateLink service.
- Tags for the VPC
- Conditions to enable/disable AWS PrivateLink service.

## Checklist

- [X] I have tested these changes thoroughly.
- [X] My code follows the project's coding style.
- [X] I have added appropriate comments to my code, especially in complex areas.
- [X] All new and existing tests passed locally.

[GDB-10116]: https://ontotext.atlassian.net/browse/GDB-10116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/vpc/variables.tf [hunk 1]
```diff
@@ -38,4 +38,24 @@ variable "single_nat_gateway" {
 variable "enable_nat_gateway" {
   description = "Enalbe or disable the creation of the NAT Gateway"
   type        = bool
-}
\ No newline at end of file
+}
+
+variable "vpc_endpoint_service_acceptance_required" {
```
Comment (viktor-ribchev) [inline modules/vpc/variables.tf:43]: This name is confusing, can be changed to `accept_vpc_endpoint_connection_requests`

File: variables.tf [hunk 2]
```diff
@@ -160,6 +160,24 @@ variable "enable_nat_gateway" {
...
+  default     = true
+}
+
+variable "vpc_endpoint_service_allowed_principals" {
+  description = "(Optional) The ARNs of one or more principals allowed to discover the endpoint service."
+  type        = list(string)
+  default     = null
+}
+
+variable "graphdb_enable_private_access" {
```
Comment (viktor-ribchev) [inline variables.tf:175]: Shouldn't this be `lb_enable_priavate_access` ? after all the private link leads to the network lb and not directly to GDB

File: variables.tf [hunk 3]
```diff
@@ -160,6 +160,24 @@ variable "enable_nat_gateway" {
   default     = true
 }
 
+variable "accept_vpc_endpoint_connection_requests" {
```
Comment (mihailradkov) [inline variables.tf:163]: `vpc_endpoint_service_accept_connection_requests` to be consistent with `vpc_endpoint_service_allowed_principals`

File: modules/graphdb/nsg.tf [hunk 4]
```diff
@@ -58,6 +58,8 @@ resource "aws_security_group_rule" "graphdb_network_lb_ingress" {
 }
 
 resource "aws_security_group_rule" "graphdb_lb_healthchecks" {
+  count = var.lb_enable_private_access ? 0 : 1 # Since it creates duplicated rule if lb_internal is true we need to have a toggle to enable/disable this rule based on the type of the access to the LB
```
Comment (mihailradkov) [inline modules/graphdb/nsg.tf:61]: Put the comment before the count. It's hard to read on narrow resolutions

File: variables.tf [hunk 5]
```diff
@@ -160,6 +160,24 @@ variable "enable_nat_gateway" {
...
+  default     = true
+}
+
+variable "vpc_endpoint_service_allowed_principals" {
+  description = "(Optional) The ARNs of one or more principals allowed to discover the endpoint service."
+  type        = list(string)
+  default     = null
+}
+
+variable "lb_enable_private_access" {
```
Comment (mihailradkov) [inline variables.tf:175]: Maybe add a short example in the readme with what should/could be configured for the private lb/access

**Review by mihailradkov — APPROVED** [review]

**Pull request merged**